src/ui/buttons/props.js

```diff
@@ -696,6 +719,56 @@ export function normalizeButtonStyle(
   };
 }
 
+export function validateButtonMessage(
+  props: ?ButtonPropsInputs,
+  message: ButtonMessageInputs
+): ButtonMessage {
+  if (!message) {
+    throw new Error(`Expected props.message to be set`);
+  }
+
+  props = props || getDefaultButtonPropsInput();
+
+  const { amount, offer, color, position, align } = message;
+
+  if (amount !== undefined) {
+    if (typeof amount !== "number") {
+      throw new TypeError(
+        `Expected style.height to be a number, got: ${amount}`
+      );
+    }
+  }
+
+  if (offer) {
```

Comment:
Should we consider checking to ensure that this is actually an array before we try to iterate over it below?

Reply:
@Seavenly while we're on the topic of `offer`'s type, should we consider converting a string value here to an array? Might give users a helping hand if they fail to notice we're looking for an array and not just the offer name.

Reply:
I'm not sure there's precedence for that already for the buttons component so we may want to avoid that in favor of the errors pointing the user to the correct integration.